dipy/reconst/csdeconv.py

```diff
@@ -798,3 +839,114 @@ def auto_response(gtab, data, roi_center=None, roi_radius=10, fa_thr=0.7,
         return response, ratio, indices[0].size
 
     return response, ratio
+
+
+def recursive_response(gtab, data, mask=None, sh_order=8, peak_thr=0.01,
+                       init_fa=0.08, init_trace=0.0021, iter=8,
+                       convergence=0.001, parallel=True,
+                       sphere=default_sphere):
+    """ Recursive calibration of response function using peak threshold
+
+    Parameters
+    ----------
+    gtab : GradientTable
+    data : ndarray
+        diffusion data
+    mask : ndarray
+        mask for recursive calibration, for example a white matter mask. It has
+        shape `data.shape[0:3]` and dtype=bool.
+    sh_order : int
+        maximal spherical harmonics order
+    peak_thr : float
+        peak threshold, how large the second peak can be relative to the first
+        peak in order to call it a single fiber population [1]
+    init_fa : float
+        FA of the initial 'fat' response function (tensor)
+    init_trace : float
+        trace of the initial 'fat' response function (tensor)
+    iter : int
+        maximum number of iterations for calibration
+    convergence : float
+        convergence criterion, maximum relative change of SH coefficients
+    sphere : Sphere
```

Comment:
Yup.

On Sat, Mar 14, 2015 at 2:38 PM, Samuel St-Jean <notifications@github.com>
wrote:

> In dipy/reconst/csdeconv.py
> <https://github.com/nipy/dipy/pull/588#discussion_r26442003>:
>
> > +        mask for recursive calibration, for example a white matter mask. It has
> > +        shape `data.shape[0:3]` and dtype=bool.
> > +    sh_order : int
> > +        maximal spherical harmonics order
> > +    peak_thr : float
> > +        peak threshold, how large the second peak can be relative to the first
> > +        peak in order to call it a single fiber population [1]
> > +    init_fa : float
> > +        FA of the initial 'fat' response function (tensor)
> > +    init_trace : float
> > +        trace of the initial 'fat' response function (tensor)
> > +    iter : int
> > +        maximum number of iterations for calibration
> > +    convergence : float
> > +        convergence criterion, maximum relative change of SH coefficients
> > +    sphere : Sphere
>
> It could be also useful to specify the number of cores to use, as
> peaks_from_model can consider the number of processes to launch.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/588/files#r26442003>.
>